spring-cloud-azure-samples/spring-cloud-azure-storage-queue-sample/src/main/java/example/WebController.java

```diff
@@ -6,88 +6,43 @@
+
+    private static final Log LOGGER = LogFactory.getLog(WebController.class);
     private static final String STORAGE_QUEUE_NAME = "example";
-    private static final String OUTPUT_CHANNEL = "outputChannel";
-    private static final String INPUT_CHANNEL = "inputChannel";
-    private static final Log LOGGER = LogFactory.getLog(StorageQueueApplication.class);
 
     @Autowired
-    StorageQueueOutboundGateway storageQueueOutboundGateway;
-
-    /** Message gateway binding with {@link MessageHandler}
-     *  via {@link MessageChannel} has name {@value OUTPUT_CHANNEL}
-     */
-    @MessagingGateway(defaultRequestChannel = OUTPUT_CHANNEL)
-    public interface StorageQueueOutboundGateway {
-        void send(String text);
-    }
-
-    /**
-     * Posts a message to a Azure Storage Queue
-     */
-    @PostMapping(value = "/messages")
-    public void send(@RequestBody String data) {
-        storageQueueOutboundGateway.send(data);
-    }
-
-    @Bean
-    @ServiceActivator(inputChannel = OUTPUT_CHANNEL)
-    public MessageHandler messageSender(StorageQueueOperation storageQueueOperation) {
-        AzureMessageHandler handler = new AzureMessageHandler(STORAGE_QUEUE_NAME, storageQueueOperation);
-        handler.setSendCallback(new ListenableFutureCallback<Void>() {
-            @Override
-            public void onSuccess(Void result) {
-                LOGGER.info("Message was sent successfully.");
-            }
-
-            @Override
-            public void onFailure(Throwable ex) {
-                LOGGER.info("There was an error sending the message.");
-            }
-        });
-        return handler;
-    }
+    StorageQueueOperation storageQueueOperation;
 
-    @Bean
-    @InboundChannelAdapter(channel = INPUT_CHANNEL, poller = @Poller(fixedDelay = "5000"))
-    public StorageQueueMessageSource StorageQueueMessageSource(StorageQueueOperation storageQueueOperation) {
-        storageQueueOperation.setVisibilityTimeoutInSeconds(10);
-        storageQueueOperation.setCheckpointMode(CheckpointMode.RECORD);
-        storageQueueOperation.setMessagePayloadType(String.class);
-        StorageQueueMessageSource messageSource =
-                new StorageQueueMessageSource(STORAGE_QUEUE_NAME, storageQueueOperation);
-        return messageSource;
+    @PostMapping("/messages")
+    public String send(@RequestParam("message") String message) {
+        this.storageQueueOperation.sendAsync(STORAGE_QUEUE_NAME, MessageBuilder.withPayload(message).build());
+        return message;
     }
 
-    /** This message receiver binding with {@link StorageQueueMessageSource}
-     *  via {@link MessageChannel} has name {@value INPUT_CHANNEL}
-     */
-    @ServiceActivator(inputChannel = INPUT_CHANNEL)
-    public void messageReceiver(Message<?> message) {
-        LOGGER.info("message received: " + message.getPayload());
+    @GetMapping("/messages")
+    public void receive() throws ExecutionException, InterruptedException {
```

Comment:
If you want to expose this as rest api, you need to return message, rather than void.